File: week 2/readability/readability.c
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>

float count_letters(string text);
float count_words(string text);
float count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    float letters_count = count_letters(text);
    float words_count = count_words(text);
    float sentences_count = count_sentences(text);

    // Average letters per 100 words
    float L = (letters_count / words_count) * 100;

    //Average sentences per 100 words
    float S = (sentences_count / words_count) * 100;

    //Coleman-Liau index
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    //Prints the grade of the text
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}


//Counts the number of letters
float count_letters(string text)
{
    float counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        // Check for alphabet
        if (isalpha(text[i]))
        {
            counter++;
        }
    }

    return counter;
}


// Counts the number of words
float count_words(string text)
{
    float counter = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            counter++;
        }
    }

    // Because there is spaces + 1 word in a sentence
    return counter + 1;
}


// Counts the number of sentences
float count_sentences(string text)
{
    float counter = 0;

    for (int i = 0; i < strlen(text); i++)
    {
        // Check for punctuation marks
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            counter++;
        }
    }

    return counter;
}

File: week 6/sentimental/readability/readability.py
import re
from cs50 import get_string


def main():
    text = get_string("Text: ")

    # Counts the number of letters
    letters = len(re.findall("[a-zA-Z]", text))

    # Counts the number of words
    words = len(text.split())

    # Counts the number of sentences
    sentences = len(re.findall(r"[.!?]+", text))

    index = readability_index(letters, words, sentences)

    print_grade(index)


def readability_index(letters, words, sentences):
    words_per_100 = words / 100

    # Average letters per 100 words
    l = letters / words_per_100

    # Average sentences per 100 words
    s = sentences / words_per_100

    # Coleman-Liau index
    index = round(0.0588 * l - 0.296 * s - 15.8)

    return index


# Prints the grade of the text
def print_grade(index):
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


if __name__ == "__main__":
    main()
